app/src/main/java/org/mozilla/fenix/wallpapers/WallpapersUseCases.kt

```diff
@@ -0,0 +1,246 @@
+ *
+ * @param context Used for various file and configuration checks.
+ * @param store Will receive dispatches of metadata updates like the currently selected wallpaper.
+ * @param client Handles downloading wallpapers and their metadata.
+ * @param strictMode Required for determining some device state like current locale and file paths.
+ *
+ * @property initialize Usecase for initializing wallpaper feature. Should be called eagerly.
+ * @property loadBitmap Usecase for loading specific wallpaper bitmaps.
+ * @property selectUseCase Usecase for selecting a new wallpaper.
+ */
+class WallpapersUseCases(
+    context: Context,
+    store: AppStore,
+    client: Client,
+    strictMode: StrictModeManager
+) {
+    val initialize: InitializeWallpapersUseCase by lazy {
+        // Required to even access context.filesDir property and to retrieve current locale
+        val (fileManager, currentLocale) = strictMode.resetAfter(StrictMode.allowThreadDiskReads()) {
+            val fileManager = WallpaperFileManager(context.filesDir)
+            val currentLocale = LocaleManager.getCurrentLocale(context)?.toLanguageTag()
+                ?: LocaleManager.getSystemDefault().toLanguageTag()
+            fileManager to currentLocale
+        }
+        val downloader = WallpaperDownloader(context, client)
+        DefaultInitializeWallpaperUseCase(
+            store = store,
+            downloader = downloader,
+            fileManager = fileManager,
+            settings = context.settings(),
+            currentLocale = currentLocale
+        )
+    }
+    val loadBitmap: LoadBitmapUseCase by lazy { DefaultLoadBitmapUseCase(context) }
+    val selectUseCase: SelectWallpaperUseCase by lazy { DefaultSelectWallpaperUseCase(context.settings(), store) }
+
+    /**
+     * Contract for usecases that initialize the wallpaper feature.
+     */
+    interface InitializeWallpapersUseCase {
+        /**
+         * Start operations that should be down during initialization, like remote metadata
+         * retrieval and determining the currently selected wallpaper.
+         */
+        suspend operator fun invoke()
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    internal class DefaultInitializeWallpaperUseCase(
+        private val store: AppStore,
+        private val downloader: WallpaperDownloader,
+        private val fileManager: WallpaperFileManager,
+        private val settings: Settings,
+        private val currentLocale: String,
+        private val possibleWallpapers: List<Wallpaper> = allWallpapers,
+    ) : InitializeWallpapersUseCase {
+
+        /**
+         * Downloads the currently available wallpaper metadata from a remote source.
+         * Updates the [store] with that metadata and with the selected wallpaper found in storage.
+         * Removes any unused promotional or time-limited assets from local storage.
+         */
+        override suspend operator fun invoke() {
```

Comment:
for reviewing context, what percentage of `invoke` is just re-located code from `WallpaperManager`?

Reply:
It should all be re-located and slightly re-arranged code from `WallpaperManager`. Outside of making a couple things explicitly `suspend`ing, this PR shouldn't change any functionality.